### PR TITLE
Fixing subjects carousels, adds correct availability to API

### PR DESCRIPTION
**Pull request by mekarpeles**
Our subject page is super broken. Availability faceting is wring. There's no lazy loading. This to be combined with @tabshaikh's #1871

### CODE REVIEW

**Comment by tabshaikh** [comment]
lgtm by what i can understand :)

File: openlibrary/macros/AvailabilityButton.html [hunk 1]
```diff
@@ -66,3 +66,7 @@
   <form>
     <input type="button" class="cta-btn--missing cta-btn" disabled value="No ebook available">
   </form>
+  $if page.ia:
+    $ daisy_url = "/ia/%s/daisy" % page.ia[0]
+    $:macros.daisy(page, url=daisy_url, msg="Print-disabled access available")
```
Comment (hornc) [inline openlibrary/macros/AvailabilityButton.html]: What about internationalisation for the availability messaging? These should be wrapped in an appropriate i18n display wrapper

File: openlibrary/plugins/worksearch/subjects.py [hunk 2]
```diff
@@ -169,6 +177,13 @@ def normalize_key(self, key):
     def process_key(self, key):
         return key
 
+def inject_availability(subject):
+    works = add_availability(subject.works)
+    for work in works:
+        ocaid = work.ia if work.ia else None
+        availability = work.get('availability', {}).get('status')
+    subject.works = works
+    return subject
```
Comment (hornc) [inline openlibrary/plugins/worksearch/subjects.py]: This probably is ok for functionality, but the naming continues the conflation we have between "subjects" and "Availability status" -- not sure if there is an easy way to improve this, either by renaming variables to be clearer, or adding comments to explain what is going on and how availability is not a real subject :)

File: openlibrary/plugins/worksearch/subjects.py [hunk 3]
```diff
@@ -240,7 +255,10 @@ def create_engine():
     sort_order = sort_options.get(sort) or sort_options['editions']
 
     engine = create_engine()
-    return engine.get_subject(key, details=details, offset=offset, sort=sort_order, limit=limit, **filters)
+    subject = engine.get_subject(
```
Comment (hornc) [inline openlibrary/plugins/worksearch/subjects.py]: Perhaps if this `subject` was named `result` like it is in `SubjectEngine.get_subject()`'s return, so subject is something like `cats`, and `result`  is a list of works with that subject? I found this confusing at first while reviewing (See above strikethrough comment)

File: openlibrary/plugins/worksearch/subjects.py (hunk 2 above)
Comment (mekarpeles) [reply openlibrary/plugins/worksearch/subjects.py]: great point, will update var name to something clearer

**Comment by mekarpeles** [comment]
got the lgtm from @hornc

**Pull request merged**